### ISSUE TITLE
should field docstring be lowered to `@doc docstring type.field` ?

### ISSUE THREAD

**Issue by Roger-luo** [JuliaLang/julia#39825]
I come across https://github.com/Roger-luo/Configurations.jl/issues/17

I realize the current way of creating the docstring binding relies on the implicit `@doc` created on top of a `struct`, which means the docstring on fields won't work if a struct is declared without docstring on top, e.g

this won't work

```julia
struct Foo
   "x field"
    x
end
```

this will work

```julia
"doc string"
struct Foo
   "x field"
    x
end
```

the documentation also didn't mention this behavior explicitly that the former won't work: https://docs.julialang.org/en/v1/manual/documentation/#Types

I guess it makes sense to make the field docstring always work for the sake of consistency, or one needs to always special handle this syntax to be consistent with Julia, it will make meta-programming easier since the AST is more consistent (by lowering the field docstring sugar to `@doc` macro calls), so I'm proposing a lowering the converts the above code to the following

```julia
struct Foo
@doc "x field" Main.Foo.x
x
end
```

tho currently the `@doc` macro seems not to support adding documentation for a field directly.

What do you think?

**Comment by StefanKarpinski** [comment]
Seems like it could be a good idea. Added triage label to discuss on Thursday.

**Comment by vtjnash** [comment]
I think it would make sense to add the implicit `@doc`, though perhaps at the top (and filter out empty docstrings later). Though generally I would expect structs should have documentation too already.
```
@doc ""
struct Foo
"x field"
x
end
```

**Comment by Roger-luo** [comment]
> I think it would make sense to add the implicit @doc, though perhaps at the top (and filter out empty docstrings later). Though generally I would expect structs should have documentation too already.

Yeah I think this also makes sense and seems simpler than my proposal, since my main complain here is about making the `Expr` object more consistent with or without field docstring. this issue itself in general is not a big deal tho.
